### PR TITLE
Adds read and write of Arrow IPC files 

### LINKED ISSUE #140: Read and write Arrow IPC
Should be pretty straightforward. https://pola-rs.github.io/polars/polars_io/ipc/index.html

### PR DESCRIPTION
**Pull request by kimjoaoun**
If approved, this Pull Requests closes #140.

This PR creates:

```
Explorer.DataFrame.read_ipc/1
Explorer.DataFrame.write_ipc/2
```

"Example":

```Elixir

# First we create an IPC file.

Explorer.Datasets.iris
|> Explorer.DataFrame.write_ipc("iris_test.ipc")

# Then we read the same IPC file into Elixir.

Explorer.DataFrame.read_ipc("iris_test.ipc")

```

### CODE REVIEW

**Comment by kimjoaoun** [comment]
@cigrainger sorry for not asking permission to do this PR 😬

File: lib/explorer/data_frame.ex [hunk 1]
```diff
@@ -127,6 +127,21 @@ defmodule Explorer.DataFrame do
     apply_impl(df, :write_parquet, [filename])
   end
 
+  @doc """
+  Reads a IPC file into a dataframe.
+  """
+  @spec read_ipc(filename :: String.t()) :: {:ok, DataFrame.t()} | {:error, term()}
+  def read_ipc(filename), do: Explorer.PolarsBackend.DataFrame.read_ipc(filename)
```
Comment (cigrainger) [inline lib/explorer/data_frame.ex:134]: Check out the other `read_` functions for setting the default backend.

File: lib/explorer/polars_backend/data_frame.ex [hunk 2]
```diff
@@ -128,6 +128,20 @@ defmodule Explorer.PolarsBackend.DataFrame do
...
+      {:error, error} -> {:error, error}
+    end
+  end
+
+  def write_ipc(%DataFrame{data: df}, filename) do
+    case Native.df_write_ipc(df, filename) do
+      {:ok, _} -> {:ok, filename}
+      {:error, error} -> {:error, error}
+    end
+  end
```
Comment (cigrainger) [inline lib/explorer/polars_backend/data_frame.ex:143]: These will need `@impl true`.

**Review by cigrainger — CHANGES_REQUESTED** [review]
Just a couple minor changes needed. Thanks for this one!

**Comment by josevalim** [comment]
So if I get this right, is this mostly a way to serialize/deserialize dataframes? What are the use cases? Maybe we should have one or two paragraphs in the docs breaking this a bit down? :)

**Comment by cigrainger** [comment]
That's exactly right @josevalim. https://arrow.apache.org/docs/format/Columnar.html#serialization-and-interprocess-communication-ipc

I think it's particularly useful with [feather](https://arrow.apache.org/docs/python/feather.html) which is used with R and Python quite a bit.

**Comment by kimjoaoun** [comment]
I've just realised that `read_ipc` and `write_ipc` might have some extra arguments, like ["with_compression", on write_ipc](https://pola-rs.github.io/polars/polars_io/ipc/struct.IpcWriter.html). Can I implement them on a new PR?

**Comment by cigrainger** [comment]
@kimjoaoun Yes please! And could you throw in a bit of explanation to handle #147?

**Comment by kimjoaoun** [comment]
Of course!

**Pull request merged**